apps/condo/domains/user/utils/serverSchema/guards.js

```diff
@@ -49,18 +78,33 @@ class RedisGuard {
         }
     }
 
+    async incrementDayCounter (variable) {
+        return this.incrementCustomCounter(variable, dayjs().endOf('day'))
+    }
+
     // Counter will reset at the start of a day ( or after redis restart )
     // Example usage = only 100 attempts to confirm phone from single IP
-    async incrementDayCounter (variable) {
+    async incrementCustomCounter (variable, date) {
         // if variable not exists - it will be set to 1
         let afterIncrement = await this.redis.incr(`${this.counterPrefix}${variable}`)
         afterIncrement = Number(afterIncrement)
         if (afterIncrement === 1) {
-            await this.redis.expireat(`${this.counterPrefix}${variable}`, parseInt(`${dayjs().endOf('day') / 1000}`))
+            await this.redis.expireat(`${this.counterPrefix}${variable}`, parseInt(`${date / 1000}`))
         }
         return afterIncrement
     }
 
+    // Counter
+    // 1. Set variable to reddis with TTL
+    // 2. Check if counter exists
+    // 3. Get counter remain time
+    async counterTimeRemain (variable) {
+        const time = await this.redis.ttl(`${this.counterPrefix}${variable}`)
+        // -1: no ttl on variable, -2: key not exists
+        return Math.max(time, 0)
+    }
```

Comment:
I thought we had this function somewhere in the code. We use RedisGuard in login procedure. I think it even has the sliding window feature, but I can be mistaken

Please, check this out

Reply:
This don't. I added this function since we have no way to get counter time remain, (you probably mean lock time remain, but that function not suitable for counters)